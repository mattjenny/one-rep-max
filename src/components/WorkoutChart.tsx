import React from 'react';
import styled from 'styled-components';
import {
    VictoryChart,
    VictoryArea,
    VictoryLine,
    VictoryScatter,
    VictoryTooltip,
    VictoryGroup,
    DomainPropType,
} from 'victory';
import { DARK_GRAY, GREEN } from '../constants/colors';
import { IWorkoutExercise } from '../store/types';
import { chartTheme } from './chartTheme';

const WorkoutChartWrapper = styled.div``;

export interface IWorkoutChartProps {
    data: IWorkoutExercise[];
    domain: DomainPropType;
}

export function WorkoutChart({ data, domain }: IWorkoutChartProps) {
    return (
        <WorkoutChartWrapper>
            <svg width="0" height="0">
                <defs>
                    <linearGradient
                        id="chartGradient"
                        x1="0%"
                        y1="0%"
                        x2="0%"
                        y2="100%"
                    >
                        <stop offset="10%" stopColor={GREEN} />
                        <stop offset="90%" stopColor={DARK_GRAY} />
                    </linearGradient>
                </defs>
            </svg>
            <VictoryChart theme={chartTheme}>
                <VictoryGroup
                    data={data}
                    domain={domain}
                    scale="time"
                    labelComponent={
                        <VictoryTooltip
                            style={{ fontSize: 10 }}
                        />
                    }
                >
                    <VictoryArea
                        style={{ data: { fill: 'url(#chartGradient)' } }}
                    />
                    <VictoryLine
                        style={{ data: { stroke: GREEN, strokeWidth: 2 } }}
                    />
                    <VictoryScatter
                        size={1}
                        style={{ data: { fill: GREEN } }}
                    />
                </VictoryGroup>
            </VictoryChart>
        </WorkoutChartWrapper>
    );
}
