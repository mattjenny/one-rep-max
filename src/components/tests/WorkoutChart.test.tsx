import { shallow } from 'enzyme';
import React from 'react';
import { WorkoutChart } from '../WorkoutChart';
import { chartData, domain } from './testConstants';

describe('WorkoutChart component', () => {
    it('renders correctly', () => {
    const wrapper = shallow(<WorkoutChart data={chartData} domain={domain} />)
    expect(wrapper.html()).toMatchSnapshot();
    });
});
