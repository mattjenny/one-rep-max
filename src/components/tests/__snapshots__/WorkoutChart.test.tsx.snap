// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutChart component renders correctly 1`] = `
"<div class=\\"sc-bdVaJa bDWFJH\\"><svg width=\\"0\\" height=\\"0\\"><defs><linearGradient id=\\"chartGradient\\" x1=\\"0%\\" y1=\\"0%\\" x2=\\"0%\\" y2=\\"100%\\"><stop offset=\\"10%\\" stop-color=\\"#4EC183\\"></stop><stop offset=\\"90%\\" stop-color=\\"#242A2E\\"></stop></linearGradient></defs></svg><div style=\\"height:100%;width:100%;user-select:none;pointer-events:none;touch-action:none;position:relative\\" class=\\"VictoryContainer\\"><svg width=\\"450\\" height=\\"300\\" role=\\"img\\" aria-labelledby=\\"victory-container-1-title victory-container-1-desc\\" viewBox=\\"0 0 450 300\\" style=\\"pointer-events:all;width:100%;height:auto\\"><g><g style=\\"height:100%;width:100%;user-select:none\\" clip-path=\\"url(#victory-clip-2)\\"><defs><clipPath id=\\"victory-clip-2\\"><rect x=\\"60\\" y=\\"20\\" width=\\"390\\" height=\\"220\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></rect></clipPath></defs><path d=\\"M60,213.19327731092437L225.1764705882353,80.08403361344546L450,46.80672268907563L450,240L225.1764705882353,240L60,240Z\\" style=\\"fill:url(#chartGradient);width:1px;stroke:url(#chartGradient)\\" role=\\"presentation\\" shape-rendering=\\"auto\\"></path></g><g style=\\"height:100%;width:100%;user-select:none\\" clip-path=\\"url(#victory-clip-3)\\"><defs><clipPath id=\\"victory-clip-3\\"><rect x=\\"60\\" y=\\"20\\" width=\\"390\\" height=\\"220\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></rect></clipPath></defs><path d=\\"M60,213.19327731092437L225.1764705882353,80.08403361344546L450,46.80672268907563\\" style=\\"fill:none;stroke:#4EC183;stroke-width:2;width:1px;opacity:1\\" role=\\"presentation\\" shape-rendering=\\"auto\\"></path></g><g><path d=\\"M 60, 213.19327731092437
      m -1, 0
      a 1, 1 0 1,0 2,0
      a 1, 1 0 1,0 -2,0\\" style=\\"fill:#4EC183;width:1px\\" role=\\"presentation\\" shape-rendering=\\"auto\\"></path><path d=\\"M 225.1764705882353, 80.08403361344546
      m -1, 0
      a 1, 1 0 1,0 2,0
      a 1, 1 0 1,0 -2,0\\" style=\\"fill:#4EC183;width:1px\\" role=\\"presentation\\" shape-rendering=\\"auto\\"></path><path d=\\"M 450, 46.80672268907563
      m -1, 0
      a 1, 1 0 1,0 2,0
      a 1, 1 0 1,0 -2,0\\" style=\\"fill:#4EC183;width:1px\\" role=\\"presentation\\" shape-rendering=\\"auto\\"></path></g></g><g role=\\"presentation\\"><line x1=\\"60\\" x2=\\"450\\" y1=\\"240\\" y2=\\"240\\" style=\\"stroke:#90A4AE;fill:transparent;stroke-width:2;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><g role=\\"presentation\\"><line x1=\\"167.8235294117647\\" x2=\\"167.8235294117647\\" y1=\\"240\\" y2=\\"20\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"167.8235294117647\\" x2=\\"167.8235294117647\\" y1=\\"240\\" y2=\\"245\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"167.8235294117647\\" dx=\\"0\\" y=\\"253\\" dy=\\"10.26\\"><tspan x=\\"167.8235294117647\\" dx=\\"0\\" text-anchor=\\"middle\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">Jun 9</tspan></text></g><g role=\\"presentation\\"><line x1=\\"277.94117647058823\\" x2=\\"277.94117647058823\\" y1=\\"240\\" y2=\\"20\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"277.94117647058823\\" x2=\\"277.94117647058823\\" y1=\\"240\\" y2=\\"245\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"277.94117647058823\\" dx=\\"0\\" y=\\"253\\" dy=\\"10.26\\"><tspan x=\\"277.94117647058823\\" dx=\\"0\\" text-anchor=\\"middle\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">Jun 11</tspan></text></g><g role=\\"presentation\\"><line x1=\\"388.05882352941177\\" x2=\\"388.05882352941177\\" y1=\\"240\\" y2=\\"20\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"388.05882352941177\\" x2=\\"388.05882352941177\\" y1=\\"240\\" y2=\\"245\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"388.05882352941177\\" dx=\\"0\\" y=\\"253\\" dy=\\"10.26\\"><tspan x=\\"388.05882352941177\\" dx=\\"0\\" text-anchor=\\"middle\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">Jun 13</tspan></text></g></g><g role=\\"presentation\\"><line x1=\\"60\\" x2=\\"60\\" y1=\\"20\\" y2=\\"240\\" style=\\"stroke:#90A4AE;fill:transparent;stroke-width:2;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><g role=\\"presentation\\"><line x1=\\"60\\" x2=\\"450\\" y1=\\"235.3781512605042\\" y2=\\"235.3781512605042\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"60\\" x2=\\"55\\" y1=\\"235.3781512605042\\" y2=\\"235.3781512605042\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"47\\" dx=\\"0\\" y=\\"235.3781512605042\\" dy=\\"4.26\\"><tspan x=\\"47\\" dx=\\"0\\" text-anchor=\\"end\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">120 lbs</tspan></text></g><g role=\\"presentation\\"><line x1=\\"60\\" x2=\\"450\\" y1=\\"181.76470588235296\\" y2=\\"181.76470588235296\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"60\\" x2=\\"55\\" y1=\\"181.76470588235296\\" y2=\\"181.76470588235296\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"47\\" dx=\\"0\\" y=\\"181.76470588235296\\" dy=\\"4.26\\"><tspan x=\\"47\\" dx=\\"0\\" text-anchor=\\"end\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">130 lbs</tspan></text></g><g role=\\"presentation\\"><line x1=\\"60\\" x2=\\"450\\" y1=\\"128.15126050420173\\" y2=\\"128.15126050420173\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"60\\" x2=\\"55\\" y1=\\"128.15126050420173\\" y2=\\"128.15126050420173\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"47\\" dx=\\"0\\" y=\\"128.15126050420173\\" dy=\\"4.26\\"><tspan x=\\"47\\" dx=\\"0\\" text-anchor=\\"end\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">140 lbs</tspan></text></g><g role=\\"presentation\\"><line x1=\\"60\\" x2=\\"450\\" y1=\\"74.53781512605048\\" y2=\\"74.53781512605048\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"60\\" x2=\\"55\\" y1=\\"74.53781512605048\\" y2=\\"74.53781512605048\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"47\\" dx=\\"0\\" y=\\"74.53781512605048\\" dy=\\"4.26\\"><tspan x=\\"47\\" dx=\\"0\\" text-anchor=\\"end\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">150 lbs</tspan></text></g><g role=\\"presentation\\"><line x1=\\"60\\" x2=\\"450\\" y1=\\"20.924369747899192\\" y2=\\"20.924369747899192\\" style=\\"stroke:rgba(255, 255, 255, 0.6);fill:none;stroke-linecap:round;stroke-linejoin:round;pointer-events:painted\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><line x1=\\"60\\" x2=\\"55\\" y1=\\"20.924369747899192\\" y2=\\"20.924369747899192\\" style=\\"stroke:#90A4AE;fill:transparent;size:5px;stroke-width:1;stroke-linecap:round;stroke-linejoin:round\\" role=\\"presentation\\" shape-rendering=\\"auto\\" vector-effect=\\"non-scaling-stroke\\"></line><text direction=\\"inherit\\" x=\\"47\\" dx=\\"0\\" y=\\"20.924369747899192\\" dy=\\"4.26\\"><tspan x=\\"47\\" dx=\\"0\\" text-anchor=\\"end\\" style=\\"font-family:&#x27;Roboto&#x27;, &#x27;Helvetica Neue&#x27;, Helvetica, sans-serif;font-size:12px;letter-spacing:normal;padding:8px;fill:#FFFFFF;stroke:transparent;stroke-width:0\\">160 lbs</tspan></text></g></g></svg><div style=\\"z-index:99;position:absolute;top:0;left:0;width:100%;height:auto\\"><svg width=\\"450\\" height=\\"300\\" viewBox=\\"0 0 450 300\\" style=\\"overflow:visible;width:100%;height:auto\\"></svg></div></div></div>"
`;
